src/index.ts

```diff
@@ -60,8 +60,9 @@ const getOptionalInput = (name: string) => core.getInput(name) || undefined;
         console.log("Found existing user .npmrc file");
         const userNpmrcContent = await fs.readFile(userNpmrcPath, "utf8");
         const authLine = userNpmrcContent.split("\n").find((line) => {
-          // check based on https://github.com/npm/cli/blob/8f8f71e4dd5ee66b3b17888faad5a7bf6c657eed/test/lib/adduser.js#L103-L105
-          return /^\s*\/\/registry\.npmjs\.org\/:[_-]authToken=/i.test(line);
+          // npm Registry: check based on https://github.com/npm/cli/blob/8f8f71e4dd5ee66b3b17888faad5a7bf6c657eed/test/lib/adduser.js#L103-L105 
+          // GitHub Packages: check based on https://docs.github.com/en/packages/working-with-a-github-packages-registry/working-with-the-npm-registry#authenticating-with-a-personal-access-token
+          return /^\s*\/\/registry\.npmjs\.org\/:[_-]authToken=/i.test(line) || /^\s*\/\/npm\.pkg\.github\.com\/:[_-]authToken=/i.test(line)
```

Comment:
Shouldn't the regex here not be registry-specific but rather look for something more generic like just the `:_authToken=` part?